From 9f5be5ada9ad7f60a3267f49e0948f406f32ceb3 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Thu, 8 Aug 2019 19:21:40 +0800
Subject: [PATCH] Rule accounts_password_pam_ucredit cracklib_accounts_password_pam_ucredit add oval check

For rule cracklib_accounts_password_pam_ucredit add oval check

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../bash/wrlinux1019.sh                            | 14 +++++++++
 .../oval/wrlinux1019.xml                           | 36 ++++++++++++++++++++++
 2 files changed, 50 insertions(+)
 create mode 100644 linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_ucredit/bash/wrlinux1019.sh
 create mode 100644 linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_ucredit/oval/wrlinux1019.xml

diff --git a/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_ucredit/bash/wrlinux1019.sh b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_ucredit/bash/wrlinux1019.sh
new file mode 100644
index 0000000..2d1dc66
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_ucredit/bash/wrlinux1019.sh
@@ -0,0 +1,14 @@
+# platform = multi_platform_wrlinux
+
+. /usr/share/scap-security-guide/remediation_functions
+populate var_password_pam_ucredit
+
+if grep -q '^[^#].*pam_cracklib.so' /etc/pam.d/common-password; then
+  if grep -q 'ucredit=' /etc/pam.d/common-password; then
+	  sed -i "s/ucredit=[0-9]*/ucredit=$var_password_pam_ucredit/" /etc/pam.d/common-password
+  else
+	  sed -i "/password.*pam_cracklib.so/s/$/ ucredit=$var_password_pam_ucredit/" /etc/pam.d/common-password
+  fi
+else
+  echo "password  required pam_cracklib.so ucredit=$var_password_pam_ucredit" >>/etc/pam.d/common-password
+fi
diff --git a/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_ucredit/oval/wrlinux1019.xml b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_ucredit/oval/wrlinux1019.xml
new file mode 100644
index 0000000..aeb339b
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_ucredit/oval/wrlinux1019.xml
@@ -0,0 +1,36 @@
+<def-group>
+  <definition class="compliance" id="cracklib_accounts_password_pam_ucredit" version="1">
+    <metadata>
+      <title>Set Password ucredit Requirements</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The password ucredit should meet minimum requirements using pam_cracklib</description>
+    </metadata>
+    <criteria>
+      <criterion comment="Conditions for ucredit are satisfied" test_ref="test_password_pam_cracklib_ucredit" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" comment="check the configuration of /etc/pam.d/common-password" id="test_password_pam_cracklib_ucredit" version="1">
+    <ind:object object_ref="obj_password_pam_cracklib_ucredit" />
+    <ind:state state_ref="state_password_pam_cracklib_ucredit" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="obj_password_pam_cracklib_ucredit" version="1">
+    <ind:path>/etc/pam.d</ind:path>
+    <ind:filename>common-password</ind:filename>
+    <ind:pattern operation="pattern match">^password[\s]+(?:requisite|required)[\s]+(?:/lib/security/(?:\$ISA/)*)*pam_cracklib.so[\w_\.\-=\s]*[\s]+ucredit=(-?\d+)(?:[\s]|$)</ind:pattern>
+    <ind:instance datatype="int" operation="less than or equal">1</ind:instance>
+  </ind:textfilecontent54_object>
+  <ind:textfilecontent54_state id="state_password_pam_cracklib_ucredit" version="1">
+    <ind:instance datatype="int">1</ind:instance>
+    <ind:subexpression datatype="int"
+    operation="less than or equal"
+    var_ref="var_password_pam_ucredit" />
+  </ind:textfilecontent54_state>
+
+  <external_variable comment="External variable for pam_cracklib ucredit"
+  datatype="int" id="var_password_pam_ucredit"
+  version="1" />
+
+</def-group>
-- 
1.9.1

